src/app/Console/Commands/RequireDevTools.php

```diff
@@ -0,0 +1,157 @@
+                    }
+
+                    File::put('auth.json', json_encode($authFile, JSON_PRETTY_PRINT));
+                }
+                $this->progressBar->advance();
+            });
+        }
+
+        // Check if repositories exists
+        $details = null;
+        $process = new Process(['composer', 'config', 'repositories.backpack/devtools']);
+        $process->run(function ($type, $buffer) use (&$details) {
+            if ($type !== Process::ERR && $buffer !== '') {
+                $details = json_decode($buffer);
+            } else {
+                $details = json_decode(File::get('composer.json'), true)['repositories']['backpack/devtools'] ?? false;
+            }
+            $this->progressBar->advance();
+        });
+
+        // Create repositories
+        if (! $details) {
+            $this->info(' Creating repositories entry in composer.json');
+
+            $process = new Process(['composer', 'config', 'repositories.backpack/devtools', 'composer', 'https://repo.backpackforlaravel.com']);
+            $process->run(function ($type, $buffer) {
+                if ($type === Process::ERR) {
+                    // Fallback
+                    $composerJson = Str::of(File::get('composer.json'));
+
+                    $currentRepositories = json_decode($composerJson, true)['repositories'] ?? false;
+
+                    $repositories = Str::of(json_encode([
+                        'repositories' => [
+                            'backpack/devtools' => [
+                                'type' => 'composer',
+                                'url' => 'https://repo.backpackforlaravel.com',
+                            ],
+                        ],
+                    ], JSON_PRETTY_PRINT | JSON_UNESCAPED_SLASHES));
+
+                    $composerJson = $currentRepositories
+                        // Replace current repositories
+                        ? preg_replace('/"repositories":\s{\r?\n/', $repositories->match("/\{\n\s+([\s\S]+)\n\s{4}\}/m")->append(",\n"), $composerJson)
+                        // Append to the end of the file
+                        : preg_replace("/\r?\n}/", $repositories->replaceFirst('{', ','), $composerJson);
+
+                    File::put('composer.json', $composerJson);
+                }
+                $this->progressBar->advance();
+            });
+        }
+
+        // Require package
+        $process = new Process(['composer', 'require', '--dev', '--with-all-dependencies', 'backpack/devtools']);
+        $process->run(function ($type, $buffer) {
+            $this->progressBar->advance();
+        });
+
+        // Finish
+        $this->progressBar->finish();
+        $this->info(' DevTools is now required.');
+
+        // DevTools inside installer
+        $this->info('');
+        $this->info(' Now running the DevTools installation command.');
+
+        app()->registerDeferredProvider(\Backpack\DevTools\DevToolsServiceProvider::class);
```

Comment:
```suggestion
        app()->boot();
```

This didn't work for me - it failed with 

<img width="1014" alt="Screenshot 2021-09-17 at 12 31 30" src="https://user-images.githubusercontent.com/1032474/133761185-82070ae4-3fd4-4b29-aaf5-0faf93dd9c19.png">

How about if we boot the entire app again here? @pxpm ?

Reply:
Alternatively, if the problem is composer not having auto-loaded DevTools, just because it was installed _in the same request_, what if:
- after we do the `composer require` command, we check that the command "had effects":
    - is `backpack/devtools` now in `composer.json`?
    - is `backpack/devtools` now in `composer.lock`?
    - is devtools now in `bootstrap/cache/providers.php`?
    - is devtools now in `bootstrap/cache/services.php`?

I'm not saying we should do ALL of the above. Just the one that actually tells us that `devtools` was properly required and autoloaded. Then:
- if installed and autoloaded, run `backpack:devtools:install`
- if installed but NOT autoloaded, TELL the developer to run `composer dump-autoload && php artisan backpack:devtools:install` manually; it should be pretty easy, just copy-paste below and hit ENTER;

Would this be a solution? Do I understand correctly that in your case @pxpm DevTools is properly required using Composer, but because the autoload isn't refreshed, the rest doesn't work? 

PS. Also, if this happens (above) it means the dev will have more problems going forward, when using DevTools, right? After creating each Model, Migration, everything. So maybe this is a great opportunity to tell them that they need to run it as a service or something. Lead them to the docs to fix this problem for all their DevTools usage. 🤷‍♂️

Reply:
@tabacitu you are right, DevTools is properly required and installed, but the application is not aware of it. 

We will not have problems with classes and need to run as a service because: https://github.com/Laravel-Backpack/DevTools/blob/053808c618213eb13c9c41e4ae924dbea14971b8/src/Generators/BlueprintGenerator.php#L274

We manually require the file. 

With that in mind I pushed a fix for this, so I manually require the command and call it not by `string name` but by the class itself, as a string name it is not registered yet. 

![image](https://user-images.githubusercontent.com/7188159/133799928-75d92fe2-8dfe-4101-9991-7810a7a607a2.png)

@tabacitu can you please give this a go with this changes I pushed now ?

Reply:
I can confirm it works for me now!!! GREAT JOB @pxpm !!!

@promatik if you give your thumbs up too, let's merge this bad boy!!! 🎉🥳